server/etcdserver/server.go

```diff
@@ -1288,6 +1301,15 @@ func (s *EtcdServer) isLeader() bool {
 	return uint64(s.ID()) == s.Lead()
 }
 
+func (s *EtcdServer) isLeaderInRaftStateMachine() bool {
+	rs := s.raftStatus()
```

Comment:
Also, can we explain why querying `raftStatus` is thread safe?

Reply:
updated.